pkg/apis/core/validation/utils.go

```diff
@@ -195,3 +195,52 @@ func ValidateIPFamilies(ipFamilies []core.IPFamily, fldPath *field.Path) field.E
 
 	return allErrs
 }
+
+func validateDependencyWatchdogSettings(settings *core.SeedSettingDependencyWatchdog, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if settings.Endpoint != nil && settings.Weeder != nil {
+		allErrs = append(allErrs, field.Forbidden(fldPath, "only one of spec.settings.dependencyWatchdog.weeder and spec.settings.dependencyWatchdog.endpoint can be specified, not both."))
+	}
+	if settings.Probe != nil && settings.Prober != nil {
+		allErrs = append(allErrs, field.Forbidden(fldPath, "only one of spec.settings.dependencyWatchdog.prober and spec.settings.dependencyWatchdog.probe can be specified, not both."))
+
+	}
+
+	return allErrs
+}
+
+// ValidateSeedCreationWithOldDependencyWatchdogSettings validates that a seed shouldn't be created with old deprecated dependency watchdog settings
+func ValidateSeedCreationWithOldDependencyWatchdogSettings(seed *core.Seed) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if seed.Spec.Settings.DependencyWatchdog.Endpoint != nil && seed.Spec.Settings.DependencyWatchdog.Weeder == nil {
+		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec").Child("settings").Child("dependencyWatchdog").Child("endpoint"), "this field is deprecated, use field \"weeder\""))
+	}
+	if seed.Spec.Settings.DependencyWatchdog.Probe != nil && seed.Spec.Settings.DependencyWatchdog.Prober == nil {
+		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec").Child("settings").Child("dependencyWatchdog").Child("probe"), "this field is deprecated, use field \"prober\""))
+	}
```

Comment:
Nit:
```suggestion
	if seed.Spec.Settings.DependencyWatchdog.Endpoint != nil && seed.Spec.Settings.DependencyWatchdog.Weeder == nil {
		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec").Child("settings").Child("dependencyWatchdog").Child("endpoint"), fmt.Sprintf("this field is deprecated, use field %q", "weeder")))
	}
	if seed.Spec.Settings.DependencyWatchdog.Probe != nil && seed.Spec.Settings.DependencyWatchdog.Prober == nil {
		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec").Child("settings").Child("dependencyWatchdog").Child("probe"), fmt.Sprintf("this field is deprecated, use field %q", "prober")))
	}
```

Reply:
Do we need to worry about `seed.Spec.Settings.DependencyWatchdog.Prober` is nil or not, here?

Reply:
that is a seperate case just like `weeder` is replacement for `endpoint`, `prober` is replacement for `probe`